src/main/resources/pl/damianszczepanik/jenkins/buildhistorymanager/model/Rule/help-matchAtMost.html

```diff
@@ -2,12 +2,23 @@
 <p>The rule will stop being processed after a certain number of matched builds.
     Once this condition is met, only the other rules will continue to be processed.</p>
 
-<p><b>Use cases</b></p>
-<p>This feature provides the option to control the number of builds the rule can process. It may save a significant amount of time when the build history is long.</p>
+<p>Internally and in the documentation, this value is called <code>matchAtMost</code>.</p>
+
+<p>There are some special values:</p>
 <ul>
-    <li>If the value is set to <code>-1</code>, there is no limitation, all builds will be evaluated.</li>
-    <li>If the value is set to <code>0</code>, no builds will be evaluated. This disables the rule as it will not apply to any build.</li>
-    <li>If the value is set to <code>1..n</code>, once the value is reached, the rule is effectively disabled and is no longer applied.</li>
+   <li>If <code>matchAtMost = -1</code>, there is no limitation; all builds will be evaluated.</li>
+   <li>If <code>matchAtMost = 0</code>, no builds will be evaluated. This disables the rule, as it will not apply to any build.</li>
+   <li>If <code>matchAtMost > 0</code>, this rule will only be applied <code>matchAtMost</code> times 
+     in each build history manager process (in each loop over builds after a build).</li>
```

Comment:
`build history manager process` -> `plugin`, if I understand correctly meaning

Reply:
I meant the process the plugin follows after each build. Haven't looked much in the code but I guess there is some function that is executed as a hook/event from Jenkins. That function has some algorithm (process).